packages/flutter_tools/test/src/process.dart

```diff
@@ -0,0 +1,144 @@
+/// argument of a command.
+String shellEscapeCommand(List<String> command) {
+  return command.map(shellEscapeString).join(' ');
+}
+
+extension ProcessManagerExtension on ProcessManager {
+  /// Start a process and run it to completion, throwing an exception on failure.
+  ///
+  /// Like [runSync], this blocks until the child process terminates.
+  ///
+  /// If the child process exits with failure (a nonzero [ProcessResult.exitCode]),
+  /// this method throws an exception with details of the command and its output.
+  ProcessResult runSyncSuccess(
+    List<String> command, {
+    String? workingDirectory,
+    Map<String, String>? environment,
+    bool includeParentEnvironment = true,
+    // no runInShell; keep that always false
+    Encoding? stdoutEncoding = systemEncoding,
+    Encoding? stderrEncoding = systemEncoding,
+  }) {
+    final ProcessResult result = runSync(
+      command,
+      workingDirectory: workingDirectory,
+      environment: environment,
+      includeParentEnvironment: includeParentEnvironment,
+      stdoutEncoding: stdoutEncoding,
+      stderrEncoding: stderrEncoding,
+    );
+    if (result.exitCode != 0) {
+      throw Exception(
+        'child process exited with code ${result.exitCode}\n'
+        'command: ${shellEscapeCommand(command)}\n'
+        'stdout: ================================================================\n'
+        '${result.stdout}\n'
+        'stderr: ================================================================\n'
+        '${result.stderr}\n'
+        'end ====================================================================',
+      );
+    }
+    return result;
+  }
+}
+
+final RegExp _trailingNewlineRegExp = RegExp(r'\n*$');
+
+String _asShellOutput(String raw) {
+  return raw.replaceFirst(_trailingNewlineRegExp, '');
+}
+
+List<int> _bytesAsShellOutput(List<int> raw) {
+  final int end = 1 + raw.lastIndexWhere((int byte) => byte != 0x0a);
+  return end == raw.length ? raw : raw.sublist(0, end);
+}
+
+extension FileExtension on File {
+  /// Reads the file contents as a string with the semantics of `$(cat …)`,
+  /// returning null if the operation fails.
+  ///
+  /// This is commonly the intended semantics when a file was meant to be
+  /// read or written by a shell script.
+  ///
+  /// The file's contents are read using the given [Encoding], and then
+  /// any run of newlines at the end of the string is removed.
+  ///
+  /// See also:
+  /// * [ProcessResultExtension.shellOutput], for the semantics of `$(…)`
+  ///   on an arbitrary command.
+  /// * the Bash manual on command substitution `$(…)`: <https://www.gnu.org/software/bash/manual/bash.html#Command-Substitution>.
+  String? readStringLikeShell({Encoding encoding = utf8}) {
+    try {
+      return _asShellOutput(readAsStringSync(encoding: encoding));
+    } on FileSystemException {
+      return null;
+    }
+  }
+}
+
+extension ProcessResultExtension on ProcessResult {
```

Comment:
Can you make these non-extension helper methods? https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-extension